### PR TITLE
Regenerate all service NFData and Hashable instances

### PR DESCRIPTION
**Pull request by brendanhay**

### CODE REVIEW

File: lib/services/amazonka-certificatemanager/gen/Amazonka/CertificateManager/AddTagsToCertificate.hs [hunk 1]
```diff
@@ -128,9 +128,14 @@ instance Core.AWSRequest AddTagsToCertificate where
   response =
     Response.receiveNull AddTagsToCertificateResponse'
 
-instance Prelude.Hashable AddTagsToCertificate
+instance Prelude.Hashable AddTagsToCertificate where
+  hashWithSalt initialSalt AddTagsToCertificate' {..} =
+    initialSalt `Prelude.hashWithSalt` tags
```
Comment (endgame) [inline lib/services/amazonka-certificatemanager/gen/Amazonka/CertificateManager/AddTagsToCertificate.hs:133]: Looks like we're not hashing `certificateArn`? Is that right?

**Review by endgame — COMMENTED** [review]
I wonder if there's an off-by-one in the way the `Hashable` instance is generated?

File: lib/services/amazonka-certificatemanager/gen/Amazonka/CertificateManager/AddTagsToCertificate.hs (hunk 1 above)
Comment (brendanhay) [reply lib/services/amazonka-certificatemanager/gen/Amazonka/CertificateManager/AddTagsToCertificate.hs:133]: It's a deliberate compilation performance optimisation! (Will fix, thanks for catching.)

Comment (endgame) [reply lib/services/amazonka-certificatemanager/gen/Amazonka/CertificateManager/AddTagsToCertificate.hs]: `const 0` is a very fast valid hash function.

**Comment by endgame** [comment]
I suggest that for types with a single constructor, we apply `()` to `hashWithSalt` instead of `0`, to avoid the ambiguity. We might also need annotations for Ints passed into hash functions for sum types.

**Comment by endgame** [comment]
Closing this, since the generator's been fixed in another PR and I'm pretty sure I've got everything useful from here. Reopen and prod me if that's not the case.

**Pull request closed without merging**